### ISSUE TITLE
Update doc for MissingSwitchDefault

### ISSUE THREAD

**Issue by donalmurtagh** [checkstyle/checkstyle#16262]
I have read check documentation: https://checkstyle.org/checks/coding/missingswitchdefault.html#MissingSwitchDefault

I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="MissingSwitchDefault"/>
    </module>
</module>

/var/tmp $ cat Test.java
package example;

public class Test {
    public enum Status {
        ACTIVE, DISABLED
    }

    public void bug() {
        var status = Status.ACTIVE;

        switch (status) {
            case ACTIVE -> System.exit(0);
            case DISABLED -> System.exit(1);
        }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java

 [ERROR] Test.java:11:9: switch without "default" clause. [MissingSwitchDefault]
```
---

Confusion:
The `MissingSwitchDefault` check should not fail because the case statements are exhaustive and according to the document of this check

> This check does not validate any switch expressions. Rationale: The compiler requires switch expressions to be exhaustive. This means that all possible inputs must be covered.
---

Todo: we need to update documentation, see details at 
https://github.com/checkstyle/checkstyle/issues/16262#issuecomment-2635560488

**Comment by romani** [comment]
please share compile-able source:
```
$ cat Test.java 
public class Test {
    public enum Status {
        ACTIVE, DISABLED
    }

    public void bug() {
        var status = Status.ACTIVE;

        switch (status) {
            case ACTIVE -> System.exit(0);
            case DISABLED -> System.exit(1);
        }
    }
}
roman@p5510:/var/tmp$ javac Test.java 
Test.java:10: error: : expected
            case ACTIVE -> System.exit(0);
                                         ^
Test.java:11: error: : expected
            case DISABLED -> System.exit(1);
                                           ^
2 errors

```

**Comment by mahfouz72** [comment]
@romani What is your JDK version number? Because his provided code is compileable I use 22.

```
D:\CS\test cat src/Test.java
public class Test {
    public enum Status {
        ACTIVE, DISABLED
    }

    public void bug() {
        var status = Status.ACTIVE;

        switch (status) {
            case ACTIVE -> System.exit(0);
            case DISABLED -> System.exit(1);
        }
    }
}
D:\CS\test javac src/Test.java
```

@donalmurtagh 
This issue is not valid because the switch code provided is not checked for exhaustiveness by the compiler.

```
D:\CS\test cat src/Test.java
public class Test {
    public enum Status {
        ACTIVE, DISABLED, DELETED   // added new status
    }

    public void bug() {
        var status = Status.ACTIVE;

        switch (status) {
            case ACTIVE -> System.exit(0);
            case DISABLED -> System.exit(1);
        }
    }
}
D:\CS\test javac src/Test.java
```
@romani This can be closed after your review

**Comment by romani** [comment]
Thanks a lot.

@mahfouz72 , should we update https://checkstyle.org/checks/coding/missingswitchdefault.html#Example3-code to remove default from
```
case GREEN:
        System.out.println("GREEN");
        break;
      default:
        System.out.println("Something else");
    }
```

To show users that it is expected violation.

Or we can simply add case from this issue to example code, to show violation. So we will have two case, with and without violation.

**Comment by donalmurtagh** [comment]
> [@romani](https://github.com/romani) What is your JDK version number? Because his provided code is compileable I use 22.
> 

I'm using JDK 23. The code compiles without error on this version.

**Comment by romani** [comment]
>It doesn't seem helpful to require a default branch that cannot be executed.

it contradicts with whole idea of this Check, if you disagree, it is completely ok, you need disable/remove this Check from config.

>[ExhaustiveSwitchHasDefault](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#exhaustiveswitchhasdefault)

As you might noticed there are a lot of different Checks/Rules and sometime they contradict with each other, there numerous examples like this. Even in Checkstyle there are Checks that are opposite to each other, so user need to choose what he likes.

**Comment by mahfouz72** [comment]
> we can simply add case from this issue to example code, to show violation. So we will have two case, with and without violation.

Yes, we should update [example 1](https://checkstyle.org/checks/coding/missingswitchdefault.html#Example1-code) to have a code with enums (code in the issue description) and this will be violation, in example 3 we have one without violation

**Comment by romani** [comment]
One more way to reconcile users requirements, is to make Check property to not violate switches where all enums elements are used.
But here we run into checkstyle limitation , as we don't know enums declared in other classes. So new property is not an option until checkstyle starts to use jvm  class path  data.

**Comment by vishakha-mali** [comment]
Hey, is this issue completed ??

**Comment by mahfouz72** [comment]
Yes, @romani  please close the issue
